provider/core/src/resource.rs

```diff
@@ -118,6 +118,19 @@ pub struct ResourceKey {
     hash: ResourceKeyHash,
 }
 
+// I don't think this lint makes sense, see https://github.com/rust-lang/rust-clippy/issues/2627.
+// It's not possible to violate the hash invariant (k1 == k2 -> hash(k1) == hash(k2)) without
+// using non-determinism or unsafe code when PartialEq is derived.
+#[allow(clippy::derive_hash_xor_eq)]
+
+// Custom implementation to avoid hashing the path.
+impl core::hash::Hash for ResourceKey {
```

Comment:
Issue: I feel that `ResourceKeyHash` should be used instead of `ResourceKey` when a hash is needed, which is why there isn't already an `impl Hash for ResourceKey`.

Reply:
But `ResourceKeyHash` doesn't have the path, which I need for databake. This is like saying you should store `f(x)` is a hash set instead of `x`.

Reply:
I cannot approve this PR given that my PR https://github.com/unicode-org/icu4x/pull/2115 conflicts with the `Hash` impl. We can't merge both.

Reply:
What's the conflict exactly?

Reply:
We need to decide on what to do with the ResourceKeyMetadata.

1. Does it affect PartialEq? (then it needs to go into Hash)
2. Is it ignored for PartialEq? (then match statements get less efficient)
3. Does it go into the path string itself? (then it gets into Hash and PartialEq automatically)

I don't want to lock ourselves into having a Hash impl if the impl would be inefficient.

tl;dr, let's reach consensus on #2115 and then we can move on.

Reply:
This is independent of #2115. Hashing on hash (aka path) only is valid, even if PartialEq compares more fields. Hash is free to produce collisions as long as it produces equal hashes for PartialEq values. Hash collisions don't affect correctness, they're a performance problem, but that doesn't matter here because a) it's only used in datagen and b) we don't have keys that differ only by metadata anyway.

Reply:
ok.